TypeCasting.md

```diff
@@ -0,0 +1,42 @@
+# Type Casting
+
+In general, follow the [official guidelines](https://docs.swift.org/swift-book/LanguageGuide/TypeCasting.html).
+
+## Convention - Static casting
+Use keyword `as` for static casting. Only use static casting when automatic type casting is not available or when the compiler's inferred type does not match the intended type.
```

Comment:
re "Use keyword `as` for static casting" - is there any other option?

Reply:
I think the original intent was to specify doing something like `42 as CGFloat` rather than `CGFloat(42)`

Reply:
I can make it explicit to use as instead of doing CGFloat(42)

Reply:
Well in the case of CGFloat(42), the init is called so technically it's not a cast. I guess the convention here is that Swift casting syntax is different from Obj-C casting.